### PR TITLE
Use SetIfAbsent instead of Set to avoid dup sendInstanceBeat task

### PR DESCRIPTION
**Pull request by YonkaFang**

### CODE REVIEW

File: clients/naming_client/beat_reactor.go [hunk 1]
```diff
@@ -61,8 +61,9 @@ func buildKey(serviceName string, ip string, port uint64) string {
 func (br *BeatReactor) AddBeatInfo(serviceName string, beatInfo model.BeatInfo) {
 	logger.Infof("adding beat: <%s> to beat map", util.ToJsonString(beatInfo))
 	k := buildKey(serviceName, beatInfo.Ip, beatInfo.Port)
-	br.beatMap.Set(k, &beatInfo)
-	go br.sendInstanceBeat(k, &beatInfo)
+	if br.beatMap.SetIfAbsent(k, &beatInfo) {
```
Comment (sanxun0325) [inline clients/naming_client/beat_reactor.go:64]: use reflect.DeepEqual contrast beatInfo

Comment (sanxun0325) [reply clients/naming_client/beat_reactor.go:64]: @YonkaFang is there any progress?

Comment (YonkaFang) [reply clients/naming_client/beat_reactor.go:64]: I still think it'not proper to check the beatInfo-change thing here. As when beatInfo actually changes, override it without stop the relevant task will bring a goroutine leak.

Comment (YonkaFang) [reply clients/naming_client/beat_reactor.go:64]: To complete this case, it depends on the upper layer to do a check-and-then or stop-then-add action.

Comment (sanxun0325) [reply clients/naming_client/beat_reactor.go:64]: Added beatInfo comparison logic

1. If the beatInfo is exactly the same, nothing needs to be done
2. If no, check whether the current beatInfo exists in beatMap
             if exists, then RemoveBeatInfo func is executed

Comment (sanxun0325) [reply clients/naming_client/beat_reactor.go:64]: What do you think about this?
```
func (br *BeatReactor) AddBeatInfo(serviceName string, beatInfo model.BeatInfo) {
	logger.Infof("adding beat: <%s> to beat map", util.ToJsonString(beatInfo))
	k := buildKey(serviceName, beatInfo.Ip, beatInfo.Port)
	if _, ok := br.beatMap.Get(k); ok {
		br.RemoveBeatInfo(serviceName, beatInfo.Ip, beatInfo.Port)
	}
	br.beatMap.Set(k, &beatInfo)
	go br.sendInstanceBeat(k, &beatInfo)
}
```

Comment (YonkaFang) [reply clients/naming_client/beat_reactor.go:64]: Note that the get-then-remove-then-set action is not atomic. It still may cause some dup tasks/goroutines.
A sync map does not ensure that.

Comment (sanxun0325) [reply clients/naming_client/beat_reactor.go:64]: Yes, but in a normal registration instance scenario there will be no concurrent registration duplicate instances。You can add locks if you're worried about concurrency, but I don't think it's necessary

**Pull request closed without merging**